boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts

```diff
@@ -143,6 +143,19 @@ arduino_spi: &flexcomm5 {
 	dma-names = "rx", "tx";
 };
 
+&flexspi {
+	reg = <0x50134000 0x4000>, <0x18000000 DT_SIZE_M(64)>;
+	mx25um51345g: mx25um51345g@0 {
+		compatible = "nxp,imx-flexspi-nor";
+		size = <536870912>;
+		label = "MX25UM51345G";
+		reg = <2>;
+		spi-max-frequency = <200000000>;
+		status = "okay";
+		jedec-id = [c2 81 3a];
+	};
+};
```

Comment:
Move the relevant `reg` part to the flash device properties instead of the flash controller.

Similar to [PR](https://github.com/zephyrproject-rtos/zephyr/pull/34428)

Reply:
Let's get agreement on the direction in that PR first, before we change this one to follow.